src/core/random.hpp

```diff
@@ -1,40 +1,58 @@
 /*
   Copyright (C) 2010,2012,2013,2014,2015,2016 The ESPResSo project
 
-  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010 
+  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010
     Max-Planck-Institute for Polymer Research, Theory Group
-  
+
   This file is part of ESPResSo.
-  
+
   ESPResSo is free software: you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation, either version 3 of the License, or
   (at your option) any later version.
-  
+
   ESPResSo is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.
-  
+
   You should have received a copy of the GNU General Public License
-  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 #ifndef RANDOM_H
 #define RANDOM_H
 
-/** \file random.hpp 
+/** \file random.hpp
 
     A random generator
 */
 
+#include "errorhandling.hpp"
+
 #include <random>
 #include <string>
 #include <vector>
+#include <stdexcept>
 
 namespace Random {
 extern std::mt19937 generator;
 extern std::normal_distribution<double> normal_distribution;
 extern std::uniform_real_distribution<double> uniform_real_distribution;
+extern bool user_has_seeded;
+extern bool unseeded_error_thrown;
```

Comment:
This should not be global, I think the best solution is to make it a static local variable in check_user_has_seeded.

Reply:
hmm, but how to switch the user_has_seeded variable?

Reply:
I'm only talking about `unseeded_error_thrown`.